### ISSUE TITLE
Type.Omit preserves the $id of the TObject passed into it.

### ISSUE THREAD

**Issue by joshmossas** [sinclairzx81/typebox#565]
Given the following

```ts
const User = Type.Object(
    {
        id: Type.String(),
        name: Type.String(),
        email: Type.String(),
    },
    {
        $id: "User",
    },
);

const UserUpdatePayload = Type.Omit(User, [ "id" ]);
console.log(UserUpdatePayload)
```

`UserUpdatePayload` prints out the following

```ts
{
  '$id': 'User', // this has been preserved
  type: 'object',
  properties: {
    name: { type: 'string', [Symbol(TypeBox.Kind)]: 'String' },
    email: { type: 'string', [Symbol(TypeBox.Kind)]: 'String' }
  },
  required: [ 'name', 'email' ],
  [Symbol(TypeBox.Kind)]: 'Object'
}
  type: 'object',
  properties: {
    name: { type: 'string', [Symbol(TypeBox.Kind)]: 'String' },
    email: { type: 'string', [Symbol(TypeBox.Kind)]: 'String' }
  },
  required: [ 'name', 'email' ],
  [Symbol(TypeBox.Kind)]: 'Object'
}
```

So the id `User` was passed down to `UserUpdatePayload` which shouldn't be the case since the properties are not the same. You can get around this by also passing an `$id` to the `UserUpdatePayload` but this isn't ideal.

## Proposed Solution

```ts
  public Omit(schema: TSchema, unresolved: any, options: SchemaOptions = {}): any {
    const keys = KeyArrayResolver.Resolve(unresolved)
    // prettier-ignore
    return ObjectMap.Map(this.Discard(TypeClone.Type(schema), Transform), (object) => {
      if (ValueGuard.IsArray(object.required)) {
        object.required = object.required.filter((key: string) => !keys.includes(key as any))
        if (object.required.length === 0) delete object.required
      }
      for (const key of Object.getOwnPropertyNames(object.properties)) {
        if (keys.includes(key as any)) delete object.properties[key]
      }
      // delete the $id property if it exists before creating the new object
      if (object.$id) {
        delete object.$id
      }
      return this.Create(object)
    }, options)
  }
```

I have a working version [here](https://github.com/joshmossas/typebox) and added a test for the new functionality as well. Let me know if you'd like me to make a PR.

**Comment by sinclairzx81** [comment]
@joshmossas Hi! Thanks for reporting.

Yeah, This looks like an oversight to me. I had actually made some updates to discard `$id` on object mapping types (so Pick, Omit, Partial, Required), but looks like those updates never got merged in (due to some other technicalities I was looking at at the time). Interestingly the `Transform` symbol is omitted on all the object mapping types.

As for a PR, I should be ok on this one. I'll actually need to update the `Discard` function to support multiple key omit. There's a bit of work involved + ill need to get tests in for the update (there's a bit nuance to property discard so I'll want to be across it)

Will keep you posted.
Cheers
S

**Comment by sinclairzx81** [comment]
@joshmossas Heya,

The fix has been published on 0.31.8. The update applies to Required, Partial, Pick and Omit and will discard both `$id` and `Transform` symbol when mapping specifically for these types. 

Thanks again for reporting! This was a really good catch! Will close off the issue for now, but feel free to reply on this thread if you notice anything a miss relating to this functionality.
All the best!
S
